contracts/ServiceNodeContribution.sol

```diff
@@ -120,6 +120,18 @@ contract ServiceNodeContribution is Shared {
         contributeFunds(amount);
     }
 
+    function contributeOperatorFundsWithPermit(
+        uint256 amount,
+        IServiceNodeRewards.BLSSignatureParams memory _blsSignature,
+        uint256 deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) public onlyOperator {
+        IERC20Permit(address(SENT)).permit(msg.sender, address(this), amount, deadline, v, r, s);
```

Comment:
In theory this call should be able to look like `SENT.permit(...)`, I think?

Reply:
No bueno cause they're just interfaces, `SENT` is instantiated as `IERC20` which lets us use `transferFrom`, `approve` e.t.c but not `permit` since that's part of `IERC20Permit` interface.

Reply:
I've added the try-catch, it took the weekend but I get why they want it, specifically to allow the code to run incase someone front-runs, someone can permanently deny you from interacting with the contract if they copied just the permit input data and constantly front-run someone if they observe the TX in the mempool.